src/main/java/frc/robot/RobotContainer.java

```diff
@@ -71,11 +73,13 @@ private void configureButtonBindings() {
         driverController.rightBumper().whileTrue(stateMachine.changeState(RobotState.INTAKE_REVERSE)); // intake wheels rolled in reverse
 
         // Operator controls coral scoring
-        operatorController.a().onTrue(stateMachine.changeState(RobotState.SCORE_L1)); //spins manipulator after elevator changes
-        operatorController.x().onTrue(stateMachine.changeState(RobotState.SCORE_L2));
-        operatorController.y().onTrue(stateMachine.changeState(RobotState.SCORE_L3));
-        operatorController.b().onTrue(stateMachine.changeState(RobotState.SCORE_L4));
-
+        operatorController.leftBumper().onTrue(stateMachine.changeState(RobotState.INTAKE_CORAL));
+        operatorController.a().onTrue(Commands.runOnce(() -> stateMachine.setTargetScoreState(RobotState.SCORE_L1)));
+        operatorController.x().onTrue(Commands.runOnce(() -> stateMachine.setTargetScoreState(RobotState.SCORE_L2)));
+        operatorController.y().onTrue(Commands.runOnce(() -> stateMachine.setTargetScoreState(RobotState.SCORE_L3)));
+        operatorController.b().onTrue(Commands.runOnce(() -> stateMachine.setTargetScoreState(RobotState.SCORE_L4)));
```

Comment:
this is personal preference on my part but I think you separate the control logic from the state machine, i.e. store the target score state in robot container and then just pass that to the state machine once you actually want to store